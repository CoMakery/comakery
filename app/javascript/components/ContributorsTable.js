import React from 'react'
import PropTypes from 'prop-types'

class ContributorsTable extends React.Component {
  render() {
    return (
      <React.Fragment>
        <div className='table-scroll table-box contributors'>
          <table className='table-scroll' style={{width: '100%'}}>
            <tbody>
              <tr className='header-row'>
                <th>
                  Contributors
                </th>
                <th>
                  Total Tokens Awarded
                </th>
              </tr>

              {this.props.tableData.map((t) =>
                <tr className='award-row' key={t.name}>
                  <td className='contributor'>
                    <img src={t.imageUrl} className='icon avatar-img' />
                    <div className='margin-small margin-collapse inline-block'>
                      {t.name}
                      <table className='table-scroll table-box overlay' style={{display: 'none'}}>
                        <tbody>
                          <tr>
                            <th style={{paddingBottom: '20px'}}>
                              Contribution Summary
                            </th>
                          </tr>

                          {t.awards.map((a) =>
                            <tr key={a.name}>
                              <td>
                                {a.name}
                              </td>
                              <td>
                                {a.total}
                              </td>
                            </tr>
                          )}
                        </tbody>
                      </table>
                    </div>
                  </td>

                  <td className='awards-earned financial'>
                    <span className='margin-small'>{t.total}</span>
                  </td>
                </tr>
              )}
            </tbody>
          </table>
        </div>

      </React.Fragment>
    )
  }
}

ContributorsTable.propTypes = {
  tableData: PropTypes.array.isRequired
}
ContributorsTable.defaultProps = {
  tableData: []
}
export default ContributorsTable
