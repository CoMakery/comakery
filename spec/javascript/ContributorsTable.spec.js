import React from 'react'
import renderer from 'react-test-renderer'
import ContributorsTable from 'components/ContributorsTable'

describe('Contributors Table', () => {
    it('renders component correctly without props', () => {
      const component = renderer.create(<ContributorsTable/>)
      expect(component.toJSON()).toMatchSnapshot()
    })
    it('renders component correctly without revenue share', () => {
      const tableData = [
        {
          "imageUrl": "/test.png",
          "name": "Joe",
          "awards": [
           {
             "name": "Idea leading to patent",
             "total": "7,000"
           }
          ]
        },
        {
          "imageUrl": "/test2.png",
          "name": "Betsy",
          "awards": [
           {
             "name": "Bugfix",
             "total": "20,000"
           },
           {
             "name": "Contribution",
             "total": "5,333"
           }
          ]
        }
      ]
      const component = renderer.create(<ContributorsTable tableData={tableData} />)
      expect(component.toJSON()).toMatchSnapshot()
    })
    it('renders component correctly with revenue share', () => {
      const tableData = [
        {
          "imageUrl": "/test2.png",
          "name": "Betsy",
          "awards": [
           {
             "name": "Bugfix",
             "total": "20,000"
           },
           {
             "name": "Contribution",
             "total": "5,333"
           }
          ],
          "total": "49,877",
          "remaining": "49,377",
          "unpaid": "$1,348.68",
          "paid": "$15.08"
        }
      ]
      const component = renderer.create(<ContributorsTable tableData={tableData} revenueShare={true} />)
      expect(component.toJSON()).toMatchSnapshot()
    })
})