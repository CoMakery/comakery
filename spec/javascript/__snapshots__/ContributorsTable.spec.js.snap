// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributors Table renders component correctly with revenue share 1`] = `
<div
  className="table-scroll table-box contributors"
>
  <table
    className="table-scroll"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <tbody>
      <tr
        className="header-row"
      >
        <th>
          Contributors
        </th>
        <th>
          Total Tokens Awarded
        </th>
        <th>
          Unpaid Revenue Shares
        </th>
        <th>
          Unpaid Revenue Share Balance
        </th>
        <th>
          Lifetime Paid
        </th>
      </tr>
      <tr
        className="award-row"
      >
        <td
          className="contributor"
        >
          <img
            className="icon avatar-img"
            src="/test2.png"
          />
          <div
            className="margin-small margin-collapse inline-block"
          >
            Betsy
            <table
              className="table-scroll table-box overlay"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <tbody>
                <tr>
                  <th
                    style={
                      Object {
                        "paddingBottom": "20px",
                      }
                    }
                  >
                    Contribution Summary
                  </th>
                </tr>
                <tr>
                  <td>
                    Bugfix
                  </td>
                  <td>
                    20,000
                  </td>
                </tr>
                <tr>
                  <td>
                    Contribution
                  </td>
                  <td>
                    5,333
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </td>
        <td
          className="awards-earned financial"
        >
          <span
            className="margin-small"
          >
            49,877
          </span>
        </td>
        <td
          className="award-holdings financial"
        >
          <span
            className="margin-small"
          >
            49,377
          </span>
        </td>
        <td
          className="holdings-value financial"
        >
          <span
            className="margin-small"
          >
            $1,348.68
          </span>
        </td>
        <td
          className="paid hidden financial"
        >
          <span
            className="margin-small"
          >
            $15.08
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contributors Table renders component correctly without props 1`] = `
<div
  className="table-scroll table-box contributors"
>
  <table
    className="table-scroll"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <tbody>
      <tr
        className="header-row"
      >
        <th>
          Contributors
        </th>
        <th>
          Total Tokens Awarded
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Contributors Table renders component correctly without revenue share 1`] = `
<div
  className="table-scroll table-box contributors"
>
  <table
    className="table-scroll"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <tbody>
      <tr
        className="header-row"
      >
        <th>
          Contributors
        </th>
        <th>
          Total Tokens Awarded
        </th>
      </tr>
      <tr
        className="award-row"
      >
        <td
          className="contributor"
        >
          <img
            className="icon avatar-img"
            src="/test.png"
          />
          <div
            className="margin-small margin-collapse inline-block"
          >
            Joe
            <table
              className="table-scroll table-box overlay"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <tbody>
                <tr>
                  <th
                    style={
                      Object {
                        "paddingBottom": "20px",
                      }
                    }
                  >
                    Contribution Summary
                  </th>
                </tr>
                <tr>
                  <td>
                    Idea leading to patent
                  </td>
                  <td>
                    7,000
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </td>
        <td
          className="awards-earned financial"
        >
          <span
            className="margin-small"
          />
        </td>
      </tr>
      <tr
        className="award-row"
      >
        <td
          className="contributor"
        >
          <img
            className="icon avatar-img"
            src="/test2.png"
          />
          <div
            className="margin-small margin-collapse inline-block"
          >
            Betsy
            <table
              className="table-scroll table-box overlay"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <tbody>
                <tr>
                  <th
                    style={
                      Object {
                        "paddingBottom": "20px",
                      }
                    }
                  >
                    Contribution Summary
                  </th>
                </tr>
                <tr>
                  <td>
                    Bugfix
                  </td>
                  <td>
                    20,000
                  </td>
                </tr>
                <tr>
                  <td>
                    Contribution
                  </td>
                  <td>
                    5,333
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </td>
        <td
          className="awards-earned financial"
        >
          <span
            className="margin-small"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
